cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/handler/async/runnable/RemoveDisksRunnable.java

```diff
@@ -0,0 +1,168 @@
+import com.linkedin.kafka.cruisecontrol.model.Broker;
+import com.linkedin.kafka.cruisecontrol.model.ClusterModel;
+import com.linkedin.kafka.cruisecontrol.model.Disk;
+import com.linkedin.kafka.cruisecontrol.servlet.parameters.RemoveDisksParameters;
+import com.linkedin.kafka.cruisecontrol.servlet.response.OptimizationResult;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.util.Set;
+import java.util.Map;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.ArrayList;
+import java.util.concurrent.TimeoutException;
+import java.util.stream.Collectors;
+
+import static com.linkedin.kafka.cruisecontrol.servlet.handler.async.runnable.RunnableUtils.maybeStopOngoingExecutionToModifyAndWait;
+import static com.linkedin.kafka.cruisecontrol.servlet.handler.async.runnable.RunnableUtils.computeOptimizationOptions;
+import static com.linkedin.kafka.cruisecontrol.servlet.handler.async.runnable.RunnableUtils.isKafkaAssignerMode;
+import static com.linkedin.kafka.cruisecontrol.servlet.handler.async.runnable.RunnableUtils.SELF_HEALING_EXECUTION_PROGRESS_CHECK_INTERVAL_MS;
+import static com.linkedin.kafka.cruisecontrol.servlet.handler.async.runnable.RunnableUtils.SELF_HEALING_REPLICA_MOVEMENT_STRATEGY;
+import static com.linkedin.kafka.cruisecontrol.servlet.parameters.ParameterUtils.DEFAULT_START_TIME_FOR_CLUSTER_MODEL;
+
+public class RemoveDisksRunnable extends GoalBasedOperationRunnable {
+    private static final Logger LOG = LoggerFactory.getLogger(RemoveDisksRunnable.class);
+    private static final int CONCURRENT_INTER_BROKER_PARTITION_MOVEMENTS = 0;
+    private static final int MAX_INTER_BROKER_PARTITION_MOVEMENTS = 0;
+    private static final int CONCURRENT_INTRA_BROKER_PARTITION_MOVEMENTS = 1;
+    private static final int CONCURRENT_LEADER_MOVEMENTS = 1;
+    private final Map<Integer, Set<String>> _brokerIdAndLogdirs;
+
+    public RemoveDisksRunnable(KafkaCruiseControl kafkaCruiseControl,
+                               OperationFuture future,
+                               RemoveDisksParameters parameters,
+                               String uuid) {
+        super(kafkaCruiseControl, future, parameters, parameters.dryRun(), parameters.stopOngoingExecution(), false,
+                uuid, parameters::reason);
+        _brokerIdAndLogdirs = parameters.brokerIdAndLogdirs();
+    }
+
+    @Override
+    protected OptimizationResult getResult() throws Exception {
+        return new OptimizationResult(computeResult(), _kafkaCruiseControl.config());
+    }
+
+    @Override
+    protected void init() {
+        _kafkaCruiseControl.sanityCheckDryRun(_dryRun, _stopOngoingExecution);
+        _goalsByPriority = new ArrayList<>(1);
+        Goal intraBrokerDiskCapacityGoal = new IntraBrokerDiskCapacityGoal(true);
+        intraBrokerDiskCapacityGoal.configure(_kafkaCruiseControl.config().mergedConfigValues());
+        _goalsByPriority.add(intraBrokerDiskCapacityGoal);
```

Comment:
I think we can do something like this instead: 

`_goalsByPriority = Collections.singletonList(intraBrokerDiskCapacityGoal);`

Reply:
Good note. One observation, in `GoalBasedOperationRunnable` after the goal is completed, the `_goalsByPriority` list is cleared. This does not work with `Collections.singletonList` as it provides an immutable list. In the latest commit, I used `Collections.singletonList` but wrapped in an `ArrayList` constructor call so the list would be mutable.